inspirehep/modules/workflows/tasks/upload.py

```diff
@@ -26,56 +26,40 @@
 
 import os
 import itertools
-from pprint import pformat
-
 
 from flask import url_for
 
 from invenio_db import db
 
+from inspirehep.utils.record_getter import get_db_record
 from inspirehep.modules.pidstore.minters import inspire_recid_minter
 from inspirehep.modules.records.api import InspireRecord
 
 from ..utils import with_debug_logging
 
 
 @with_debug_logging
-def store_record(obj, *args, **kwargs):
-    """Create and index new record in main record space."""
-    obj.log.debug('Storing record: \n%s', pformat(obj.data))
-
-    assert "$schema" in obj.data, "No $schema attribute found!"
-
-    # Create record
-    # FIXME: Do some preprocessing of obj.data before creating a record so that
-    # we're sure that the schema will be validated without touching the full
-    # holdingpen stack.
-    record = InspireRecord.create(obj.data, id_=None)
-
-    for key in obj.files.keys:
-        with open(os.path.realpath(obj.files[key].file.uri)) as stream:
-            record.files[key] = stream
-            for attachment in itertools.chain(record.get('documents', []), record.get('figures', [])):
-                if key == attachment['key']:
-                    attachment['url'] = '/api/files/{bucket}/{key}'.format(bucket=record.files[key].bucket_id, key=key)
-                    break
-
-    # Create persistent identifier.
-    created_pid = inspire_recid_minter(str(record.id), record).pid_value
+def store_record(obj, eng):
+    """Insert or replace a record."""
+    is_update = obj.extra_data.get('is-update')
+    if is_update:
+        record = InspireRecord.get_record(obj.extra_data['head_uuid'])
+        record.clear()
+        record.update(obj.data)
 
-    # store head_uuid to store the root later
-    obj.extra_data['head_uuid'] = str(record.id)
-
-    # Commit any changes to record
+    else:
+        record = InspireRecord.create(obj.data, id_=None)
+        # Create persistent identifier.
+        created_pid = inspire_recid_minter(str(record.id), record).pid_value
+        obj.data['control_number'] = created_pid
+        # store head_uuid to store the root later
+        obj.extra_data['head_uuid'] = str(record.id)
+
+    attach_files_to_record_from_workflow(obj, record)
```

Comment:
I would use keyword params here (it's a bit confusing, as it's named `from_record_to_workflow` I'd expect the record to be the first param, but it's the other way around, so imo better be explicit).